index.js

```diff
@@ -500,160 +500,165 @@ const run = async userOptions => {
+        await fixFormFields({ page });
+
+        const routePath = route.replace(publicPath, "");
+        const filePath = path.join(destinationDir, routePath);
+        if (options.saveAs === "html") {
+          await saveAsHtml({ page, filePath, options, route });
+        } else if (options.saveAs === "png") {
+          await saveAsPng({ page, filePath, options, route });
         }
-        if (scriptTagText !== "") {
-          const scriptTag = document.createElement("script");
-          scriptTag.type = "text/javascript";
-          scriptTag.text = scriptTagText;
-          const firstScript = Array.from(document.scripts)[0];
-          firstScript.parentNode.insertBefore(scriptTag, firstScript);
+      },
+      onEnd: () => {
+        if (server) server.close();
+        if (http2PushManifest) {
+          const manifest = Object.keys(
+            http2PushManifestItems
+          ).reduce((accumulator, key) => {
+            if (http2PushManifestItems[key].length !== 0)
+              accumulator.push({
+                source: key,
+                headers: [
+                  {
+                    key: "Link",
+                    value: http2PushManifestItems[key]
+                      .map(x => `<${x.link}>;rel=preload;as=${x.as}`)
+                      .join(",")
+                  }
+                ]
+              });
+            return accumulator;
+          }, []);
+          fs.writeFileSync(
+            `${destinationDir}/http2-push-manifest.json`,
+            JSON.stringify(manifest)
+          );
         }
-      }, ajaxCache[route]);
-      delete ajaxCache[route];
-      if (options.fixInsertRule) await fixInsertRule({ page });
-      await fixFormFields({ page });
-
-      const routePath = route.replace(publicPath, "");
-      const filePath = path.join(destinationDir, routePath);
-      if (options.saveAs === "html") {
-        await saveAsHtml({ page, filePath, options, route });
-      } else if (options.saveAs === "png") {
-        await saveAsPng({ page, filePath, options, route });
       }
-    },
-    onEnd: () => {
-      if (server) server.close();
-      if (http2PushManifest) {
-        const manifest = Object.keys(
-          http2PushManifestItems
-        ).reduce((accumulator, key) => {
-          if (http2PushManifestItems[key].length !== 0)
-            accumulator.push({
-              source: key,
-              headers: [
-                {
-                  key: "Link",
-                  value: http2PushManifestItems[key]
-                    .map(x => `<${x.link}>;rel=preload;as=${x.as}`)
-                    .join(",")
-                }
-              ]
-            });
-          return accumulator;
-        }, []);
-        fs.writeFileSync(
-          `${destinationDir}/http2-push-manifest.json`,
-          JSON.stringify(manifest)
-        );
-      }
-    }
-  });
+    });
+  } catch (e) {
+    console.log('Exception occurred during crawling.', e);
```

Comment:
it should reject promise instead of exit. And exit should be done with `.catch` in `run.js`

Reply:
Also I realised I'm doing exit in the code myself. Will need to fix this

Reply:
are you thinking something like `return Promise.reject(new Error());` or `throw new Error();` in order to reject?

Reply:
`return Promise.reject(new Error());`

But I haven't decide how to do this better, code is not in the best shape at the moment. I can not promise you that this PR will be moving fast

Reply:
Understandable. If you are working on a refactor and this PR will just slow down that progress, you can close it. Otherwise I can make the change and the PR can sit until you are ready to merge in.

Reply:
I will leave PR open I have no reasons to close.

Reply:
I'm not sure what best practice is for `return Promise.reject(new Error());`. Do you leave it empty? Or is it best to pass in a message to `Error()`? Or should I be passing `e` to `Error()`?

Reply:
it should pass `e` itself. Something similar happens here https://github.com/peterbe/minimalcss/pull/65

Reply:
cool, thanks for the help & reference!